File: frontend/src/services/deployment.ts
import api from './api'

export interface DeploymentReport {
  summary: {
    analysis_period: string
    key_findings: string[]
    priority_actions: string[]
  }
  temporal_analysis: any
  target_analysis: any
  patrol_routes: any
  resource_allocation: any
  prevention_measures: any
  generated_at: string
}

export const deploymentApi = {
  getReport: async (days = 90): Promise<DeploymentReport> => {
    const response = await api.get('/deployment/report', { params: { days } })
    return response.data
  },

  getTemporalPatterns: async (days = 90): Promise<any> => {
    const response = await api.get('/deployment/temporal-patterns', { params: { days } })
    return response.data
  },

  getTargetPatterns: async (): Promise<any> => {
    const response = await api.get('/deployment/target-patterns')
    return response.data
  },

  getPatrolRoutes: async (radiusKm = 2.0): Promise<any> => {
    const response = await api.get('/deployment/patrol-routes', {
      params: { radius_km: radiusKm },
    })
    return response.data
  },

  getResourceAllocation: async (): Promise<any> => {
    const response = await api.get('/deployment/resource-allocation')
    return response.data
  },

  getPreventionMeasures: async (): Promise<any> => {
    const response = await api.get('/deployment/prevention-measures')
    return response.data
  },
}

File: backend/app/api/deployment.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from typing import Optional
from app.database import get_db
from app.services.deployment_service import DeploymentService

router = APIRouter()

@router.get("/report")
def get_deployment_report(
    days: int = 90,
    db: Session = Depends(get_db)
):
    """
    获取完整的工作部署建议报告
    基于已破获案件数据，生成预防性的工作部署建议
    """
    try:
        report = DeploymentService.generate_deployment_report(db, days)
        return report
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"生成部署报告失败: {str(e)}")

@router.get("/temporal-patterns")
def get_temporal_patterns(
    days: int = 90,
    db: Session = Depends(get_db)
):
    """获取时间模式分析"""
    return DeploymentService.analyze_temporal_patterns(db, days)

@router.get("/target-patterns")
def get_target_patterns(db: Session = Depends(get_db)):
    """获取目标模式分析"""
    return DeploymentService.analyze_target_patterns(db)

@router.get("/patrol-routes")
def get_patrol_routes(
    radius_km: float = 2.0,
    db: Session = Depends(get_db)
):
    """获取巡逻路线建议"""
    return DeploymentService.generate_patrol_routes(db, radius_km)

@router.get("/resource-allocation")
def get_resource_allocation(db: Session = Depends(get_db)):
    """获取资源配置建议"""
    return DeploymentService.generate_resource_allocation(db)

@router.get("/prevention-measures")
def get_prevention_measures(db: Session = Depends(get_db)):
    """获取预防措施建议"""
    return DeploymentService.generate_prevention_measures(db)

